### PR TITLE
samba4: update to 4.10.6

### PR DESCRIPTION
**Pull request by Andy2244**
Maintainer: me
Compile tested: arm (snapshots)
Run tested: arm (snapshots)

Description:
* update to 4.10.6
* use python3
* disable SERVER_AD_DC option (bugged atm)
* fix init script symlink smb.conf detection
* add new deps (libtasn1, libopenssl)

Note: SERVER_AD_DC is disabled, because it fails to build the "python embedded interpreter", seems like the build step tries to mix host/target libs. Need to investigate this if i have some more time.

_Also i could not get rid of the new libopenssl dependency (libcrypt), so keep this in mind._

PS: If someone could test macOS Timemachine limits, this would be great since i updated the patch to what was suggested in the bugzilla issue.

### CODE REVIEW

File: net/samba4/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=samba
-PKG_VERSION:=4.9.8
+PKG_VERSION:=4.10.6
 PKG_RELEASE:=1
 
 PKG_MAINTAINER:=Andy Walsh <andy.walsh44+github@gmail.com>
```
Comment (neheb) [inline net/samba4/Makefile]: I'd move this block below PKG_HASH

**Comment by Andy2244** [comment]
* added the `writeable` option for consistency and better readability

**Comment by neheb** [comment]
Package samba4-client is missing dependencies for the following libraries:
libreadline.so.8

https://downloads.openwrt.org/snapshots/faillogs/arc_arc700/packages/samba4/compile.txt

**Comment by Andy2244** [comment]
@neheb see #9470

**Comment by Andy2244** [comment]
@neheb @diizzyy Hey guys, i'm seeing several reports in the forum (samba, Davidc502) of broken, failing, crash loop on the 4.10.6 version?

Any idea whats going on here? I cant reproduce any if those so far, i suspect that maybe it has something to-do with upgrade of the package vs my full personal testing image build that includes samba.

Can you reproduce any of the problems?

**Pull request merged**